Riot/AppDelegate.m

```diff
@@ -177,6 +177,14 @@ The current call view controller (if any).
      */
     NSMutableDictionary <NSNumber *, NSMutableArray <NSDictionary *> *> *eventsToNotify;
 
+    /**
+     Cache for payloads received with incoming push notifications.
+     The key is the event id. The value, the payload.
+     Note: for the moment, objects in this dictionary are never removed but
+     the impact on memory is low.
```

Comment:
Why don't you flush this dictionary in `applicationWillEnterForeground:` (like we did for `incomingPushEventIds` array)?